### DISCUSSION TITLE
It is not possible to access the list of errors on submit without explicitly calling the `trigger` method before submission.

### DISCUSSION THREAD

**Discussion by Ploppy3** [react-hook-form/react-hook-form#10577] [Q&A] [answered]
### Version Number

7.45.0

### Codesandbox/Expo snack

https://codesandbox.io/s/flamboyant-nash-t4ljj6?file=/src/App.js

### Steps to reproduce

1. Open link
2. Click on submit
3. Notice that the field is not valid because it is required and therefore is focused. Though, the list of errors (available in console) is empty.

### Expected behaviour

Imagine a complex multi step form in which the required field is in a previous step and therefore not visible and not focusable.

Without the list of errors, it is not possible to determine on which step the validation error occurred.

Is the only solution to call `trigger()` to get the list of errors before and only then submit the form?

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@Ploppy3 

You can get list error by using `onError` callback of `handleSubmit` method.

For example:

```
const { handleSubmit } = useForm();

const onError = (errors) => {

}

const onSubmit = handleSubmit(onValid, onError);

return <form onSubmit={onSubmit}>
...
</form>
````

Reference: https://react-hook-form.com/docs/useform/handlesubmit

**Comment by Ploppy3** [comment]
Thank you very much.
